index.js

```diff
@@ -84,8 +84,8 @@ function parse(str, options) {
  */
 
 function serialize(name, val, options) {
-  var opt = options || {};
-  var enc = opt.encode || encode;
+  var opt = typeof options === 'object' ? options : {};
+  var enc = typeof opt.encode === 'function' ? opt.encode : encode;
```

Comment:
Added in spec to verify fall-through to default `encode` function when `{encode: 42}`